components/raftstore/src/store/worker/pd.rs

```diff
@@ -457,35 +465,34 @@ where
         let scheduler = self.scheduler.clone();
         let props = tikv_util::thread_group::current_properties();
 
+        fn is_enable_tick(timer_cnt: u64, interval: u64) -> bool {
+            interval != 0 && timer_cnt % interval == 0
+        }
         let h = Builder::new()
             .name(thd_name!("stats-monitor"))
             .spawn(move || {
                 tikv_util::thread_group::set_properties(props);
                 tikv_alloc::add_thread_memory_accessor();
                 let mut thread_stats = ThreadInfoStatistics::new();
                 while let Err(mpsc::RecvTimeoutError::Timeout) = rx.recv_timeout(tick_interval) {
-                    if timer_cnt % collect_store_infos_interval == 0 {
+                    if is_enable_tick(timer_cnt, collect_store_infos_interval) {
                         StatsMonitor::collect_store_infos(&mut thread_stats, &scheduler);
                     }
-                    if timer_cnt % load_base_split_check_interval == 0 {
+                    if is_enable_tick(timer_cnt, load_base_split_check_interval) {
                         StatsMonitor::load_base_split(
                             &mut auto_split_controller,
                             &receiver,
                             &scheduler,
                         );
                     }
-                    if timer_cnt % report_min_resolved_ts_interval == 0 {
+                    if is_enable_tick(timer_cnt, report_min_resolved_ts_interval) {
                         StatsMonitor::report_min_resolved_ts(
                             &region_read_progress,
                             store_id,
                             &scheduler,
                         );
                     }
-                    // modules timer_cnt with the least common multiple of intervals to avoid overflow
-                    timer_cnt = (timer_cnt + 1)
-                        % (load_base_split_check_interval
-                            * collect_store_infos_interval
-                            * report_min_resolved_ts_interval);
+                    timer_cnt += 1;
```

Comment:
would `timer_cnt` overflow?

Reply:
Currently running at the interval of one second, which I think uint64 is sufficient.